### DISCUSSION TITLE
Aws-sdk s3 upload CORS error cause by CacheControl Option

### DISCUSSION THREAD

**Discussion by JANDA-Developers** [aws/aws-sdk-js#4091] [General]
Confirm by changing [ ] to [x] below:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] This is an issue with version 2.x of the SDK

**version**
"aws-sdk": "^2.586.0",

**Describe the question**

My code is looks like this and when 
```
import { FileUpload } from "../types/scalars/Upload.scalar";

 new S3.ManagedUpload({
        params: {
            ACL: "public-read",
            Bucket: process.env.AWS_BUCKETNAME || "",
            Body: upload.createReadStream(),
            Key: filename || upload.filename,
            CacheControl: "max-age=604800,public,immutable",
        },
        // tags,
    }).promise();
```

If I upload this with the CacheControl option, upload works fine.

But when I load it on the client side I get a CORS error.

**If I remove CachControl, obviously the CORS error doesn't come up.**

My Bucket has CORS-Policy like this
```
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET",
            "POST"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]
```

Headers with CacheControl Param
 
[![enter image description here][1]][1]

Headers without CachControl Param

[![enter image description here][2]][2]

  [1]: https://i.stack.imgur.com/85E96.png
  [2]: https://i.stack.imgur.com/GV8ix.jpg

With CacheControl param,
My Allow headers are gone. Why is that?

**Comment by eternalharvest** [comment]
I have exact same issue.
I think this is a bug.

aws-sdk version is 2.1217.0.

````typescript
import S3 from 'aws-sdk/clients/s3';
import fetch from 'node-fetch';

const s3 = new S3({
    endpoint: process.env.S3_ENDPOINT,
    accessKeyId: process.env.S3_ACCESS_KEY,
    secretAccessKey: process.env.S3_SECRET_KEY,
    signatureVersion: 'v4',
});

async function main() {
    await s3.putBucketCors({
        Bucket: process.env.S3_BUCKET_NAME,
        CORSConfiguration: {
          CORSRules: [
              {
                  AllowedHeaders: ['*'],
                  AllowedMethods: ['GET'],
                  AllowedOrigins: ['*'],
                  ExposeHeaders: ['ETag'],
              }
          ]
        }
    }).promise();

    const result = await fetch(process.env.MYAPI_ENDPOINT);
    if (result.status === 200) {
      await s3.putObject({
        Bucket: process.env.S3_BUCKET_NAME,
        Key: 'data.json',
        ContentType: 'application/json',

        CacheControl: 'max-age=0,must-revalidate,public', // THIS CAUSE CORS ERROR ON BROWSER CLIENT!!

        Body: text,
      }).promise();
    }
}

main();
````

Workaround for browser side.
```javascript
await fetch(S3URL, { cache: 'no-cache' }); // SPECIFY cache OPTION TO 'no-cache' SO THAT ALWAYS REVALIDATE
```
